src/rvv/rvv_utf32_to.inl.cpp

```diff
@@ -0,0 +1,219 @@
+
+simdutf_warn_unused size_t implementation::convert_utf32_to_latin1(const char32_t *src, size_t len, char *dst) const noexcept {
+  result res = convert_utf32_to_latin1_with_errors(src, len, dst);
+  return res.error == error_code::SUCCESS ? res.count : 0;
+}
+
+simdutf_warn_unused result implementation::convert_utf32_to_latin1_with_errors(const char32_t *src, size_t len, char *dst) const noexcept {
+  const char32_t *const beg = src;
+  for (size_t vl; len > 0; len -= vl, src += vl, dst += vl) {
+    vl = __riscv_vsetvl_e32m8(len);
+    vuint32m8_t v = __riscv_vle32_v_u32m8((uint32_t*)src, vl);
+    long idx = __riscv_vfirst_m_b4(__riscv_vmsgtu_vx_u32m8_b4(v, 255, vl), vl);
+    if (idx >= 0)
+      return result(error_code::TOO_LARGE, src - beg + idx);
+    __riscv_vse8_v_u8m2((uint8_t*)dst, __riscv_vncvt_x_x_w_u8m2(__riscv_vncvt_x_x_w_u16m4(v, vl), vl), vl);
```

Comment:
Yes, but I tried to avoid vcompress when possible, because it's performance varies widely on current platforms.
I'd leave the code as is, but this is certainly something to reconsider once there is more hardware available.

Approximation from my measurements (vncvt is vnsrl.vi/vnsra.vi):
* [C908](https://camel-cdr.github.io/rvv-bench-results/canmv_k230/index.html): vncvt_m4+vncvt_m2: 12 cycles, vcompress_m4 36 cycles.
* [C920](https://camel-cdr.github.io/rvv-bench-results/milkv_pioneer/index.html): vncvt_m4+vncvt_m2: 6 cycles, vcompress_m4 5.4 cycles.

Reply:
Good idea, I've added a comment, and also one to the utf8 -> utf16 function, where one might also consider using vrgather instead of a arithmetic expression.